sql/plan/join.go

```diff
@@ -27,13 +28,29 @@ import (
 	"github.com/dolthub/go-mysql-server/sql/transform"
 )
 
+//go:generate stringer -type=JoinType -linecomment
+
 const (
 	inMemoryJoinKey        = "INMEMORY_JOINS"
 	inMemoryJoinSessionVar = "inmemory_joins"
 )
 
 var useInMemoryJoins = shouldUseMemoryJoinsByEnv()
 
+type JoinType uint16
+
+const (
+	UnknownJoinType   JoinType = iota // UnknownJoinType
```

Comment:
nit: generally speaking we name this kind of enum constant with the type first, so that all the possible values sort next to eachother